indexer/src/test/java/au/org/aodn/esindexer/utils/GeometryUtilsTest.java

```diff
@@ -0,0 +1,83 @@
+package au.org.aodn.esindexer.utils;
+
+import au.org.aodn.metadata.iso19115_3_2018.MDMetadataType;
+import jakarta.xml.bind.JAXBException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.locationtech.jts.geom.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+import static au.org.aodn.esindexer.BaseTestClass.readResourceFile;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class GeometryUtilsTest {
```

Comment:
I think should have test cases for important methods separately, likely the ones with `protected` attribute. this test is quite general given there are quite a few in between steps, e.g `breakLargeGeometryToGrid` and `removeLandAreaFromGeometry`, **and** (not just this) test specific scenarios to guarantee this kind of problem (understood that the utils will use interior point if centroid is outside the polygons, but need tests).

![image](https://github.com/user-attachments/assets/7548ef6e-7bf4-4e3b-be24-5af5fdc32f83)

Reply:
Added